R/vertica.udx.util.R

```diff
@@ -0,0 +1,92 @@
+#x is a data.frame
+#y is a list of which the first and only element is a function to run
+
+
+possible_output_types = c("boolean","int","float","real","char","varchar","longvarchar","date","datetime","smalldatetime","time","timestamp","timestamptz","timetz","numeric","varbinary","longvarbinary","binary")
+
+vertica_R_mapping <- function(type)
+{
+	if(type == "double")
+		return ("float")
+	if(type == "integer")
+		return ("int")
+	if(type == "character")
+		return ("varchar")
+}
+
+
+function_charsize = 30*1000*1000
+
+
+### the main udx execution fuction that does the work
+### receives 2 parameters params and outcols
+### params are parameters for the function
+### outcols are the output column types of the function
+
+vertica_function_executor <- function(x,y)
+{
+	parameters = deserialize_object(y$params)
+
+	#extract the function
+	vertica.dplyr.func = parameters[["vertica.dplyr.func"]]
+	parameters["vertica.dplyr.func"] <- NULL
+
+	#extract and load the necessary libraries
+	libraries = parameters[["vertica.dplyr.libraries"]]
+	parameters["vertica.dplyr.libraries"] <- NULL
+	if(length(libraries) > 0)
+		sapply(libraries, function(x) library(x,character.only = TRUE))
+
+	list2env(parameters, environment(vertica.dplyr.func))
+	
+     	output = vertica.dplyr.func(x)
+	return(output)
+}
+
+
+vertica_function_executor.OutputCallback <- function(x,y)
+{
+	outcols = deserialize_object(y$outcols)
```

Comment:
this function doesn't get access to the rows of the table. vertica will call this function before loading the table to the UDX i think (at least thats what happens in c++ it might be different in R)